include/fam/openfam_wrappers.h

```diff
@@ -0,0 +1,288 @@
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * See https://spdx.org/licenses/BSD-3-Clause
+ *
+ */
+#ifndef OPENFAM_WRAPPERS_H_
+#define OPENFAM_WRAPPERS_H_
+
+#include <stdint.h>
+#include <sys/stat.h>
+
+#ifndef int128_t
+#ifdef __int128
+typedef __int128 int128_t;
+#else
+#ifdef __SIZEOF_INT128__
+typedef __int128 int128_t;
+#else
+typedef long long int128_t;
+#endif // __SIZEOF_INT128__
+#endif // __int128
+#endif // int128_t
+
+typedef void c_fam;
+typedef void c_fam_region_desc;
+typedef void c_fam_desc;
+typedef void c_fam_context;
+
+typedef struct {
+    char *defaultRegionName;
+    char *cisServer;
+    char *grpcPort;
+    char *libfabricProvider;
+    char *famThreadModel;
+    char *cisInterfaceType;
+    char *openFamModel;
+    char *famContextModel;
+    char *numConsumer;
+    char *runtime;
+    char *fam_default_memory_type;
+    char *if_device;
+} c_fam_options;
+
+typedef enum { 
+    INTERLEAVE_DEFAULT = 0, 
+    ENABLE, 
+    DISABLE 
+} c_fam_Interleave_Enable;
+
+typedef enum { 
+    MEMORY_TYPE_DEFAULT = 0, 
+    VOLATILE, 
+    PERSISTENT 
+} c_fam_Memory_Type;
+
+typedef enum { 
+    REDUNDANCY_LEVEL_DEFAULT = 0, 
+    NONE, 
+    RAID1, 
+    RAID5 
+} c_fam_Redundancy_Level;
+
+typedef struct {
+    c_fam_Redundancy_Level redundancyLevel;
+    c_fam_Memory_Type memoryType;
+    c_fam_Interleave_Enable interleaveEnable;
+} c_fam_Region_Attributes;
+
+typedef struct {
+    uint64_t size;
+    mode_t perm;
+    char* name;
+    uint32_t uid;
+    uint32_t gid;
+    c_fam_Region_Attributes region_attributes;
+    uint64_t num_memservers;
+    uint64_t interleaveSize;
+    uint64_t memory_servers[256];
+} c_fam_stat;
+
+typedef struct {
+    uint32_t backup_option_reserved;
+} c_fam_backup_options;
+
+c_fam* c_fam_create(void);
+int  c_fam_initialize(c_fam*, const char*, c_fam_options*);
+int c_fam_initialize_ctx(c_fam*, const char*, c_fam_options*);
+int  c_fam_finalize(c_fam*, const char*);
+int c_fam_finalize_ctx(c_fam*, const char*);
```

Comment:
As discussed, having a ctx variant of the control path api to return error message when called with context object will not solve the problem. We need a mechanism to validate the object being passed.